components/omega/src/base/Halo.cpp

```diff
@@ -0,0 +1,1387 @@
+//===-- base/Halo.cpp - halo exchange methods -------------------*- C++ -*-===//
+//
+// The Halo class and its nested classes Neighbor and ExchList contain all
+// the information and methods needed for exchanging the halo elements of
+// supported YAKL array types for a given machine environment (MachEnv)
+// and parallel decomposition (Decomp). These exchanges are carried out
+// via non-blocking MPI library routines. Constructor and private member
+// functions are defined here. The Halo class public member function
+// exchangeFullArrayHalo which is called by the user to perform halo
+// exchanges on a given array is a template function and thus is defined
+// in the associated header file, Halo.h.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Halo.h"
+#include "mpi.h"
+#include <algorithm>
+#include <numeric>
+
+namespace OMEGA {
+
+// -----------------------------------------------------------------------------
+// Local routine that searches a std::vector<I4> for a particular entry and
+// returns the index of that entry. If not found, the size is returned
+// (corresponding to the last index + 1)
+
+I4 srchVector(const std::vector<I4> &InVector, // vector to search
```

Comment:
Intel compiler (correctly) flagged this as a duplicate of the same function in Decomp, so we can either rename them, protect them with a namespace or break these out into a utility module somewhere.

Reply:
Yeah, I was gonna bring this up. It would probably be a good idea to have a utility module for things like this.